Build instructions for Android (#15)
BabylonJS/JsRuntimeHost: README.md

JavaScript engine with a consistent interface. This library also provides some optional
polyfills that consumers can include if required.

## **Building - All Development Platforms**

**Required Tools:** [git](https://git-scm.com/), [CMake](https://cmake.org/), [node.js](https://nodejs.org/en/)

The first step for all development environments and targets is to clone this repository. 

Use a git-enabled terminal to follow the steps below. The `--recursive` flag is necessary as 
Babylon Native makes extensive use of submodules for dependencies.

```
git clone --recursive https://github.com/BabylonJS/JsRuntimeHost.git

> ```

Reviewer: JsRuntimeHost doesn't use submodules.

> Babylon Native requires Babylon.js. You will need to install NPM packages to resolve these dependencies. From the root of the `JsRuntimeHost` 
> repository on the command line, run the following commands:

Reviewer: I hope to remove this step some day. But in the meantime, this text is still off. :)

```suggestion
The unit tests require some NPM packages. From the root of the repository on the command line, run the following commands:
```

```
cd Tests
npm install
```

## **Building on Windows, Targeting Android**

_Follow the steps from [All Development Platforms](#all-development-platforms) before proceeding._

**Required Tools:**
[Android Studio](https://developer.android.com/studio), [Node.js](https://nodejs.org/en/download/), [Ninja](https://ninja-build.org/)

The minimal requirement target is Android 5.0.

Only building with Android Studio is supported. CMake is not used directly. Instead, Gradle
is used for building and CMake is automatically invocated for building the native part.
An `.apk` that can be executed on your device or simulator is the output.

First, download the latest release of Ninja, extract the binary, and add it to your system path.

> Once you have Android Studio downloaded, open it in Administrator Mode. Then, you need to set up an Android emulator if you do not have a physical Android device. You can do this by selecting `Tools` -> `Device Manager` and then selecting a device. (We are using Pixel 2 API 27). 

Reviewer: > open it in Administrator Mode

I don't think this is necessary?

Author: It does seem to be necessary the first time you build.

Open the project located at
`JsRuntimeHost\Tests\UnitTests\Android` with Android Studio (make sure Android Studio is in admin mode). Note that this can take a while to load. (The bottom right corner of the Android Studio window shows you what is currently being loaded.) 

Then in the LEFT PANE, right click on `app`, and select `Run 'All Tests'`. If you don't have an Android device plugged in or no Android image in the Android emulator,

> that option will be greyed and inaccessible. 

Reviewer: May be good to include a screenshot here.

**Troubleshooting:**

> If the `app/cpp` folder on the left navigation pane is empty, select `File` -> `Sync project with gradle files` and try to re-run the project by selecting  `Run` -> `Run 'All Tests'`.

Reviewer: ```suggestion
If the `app\cpp` folder on the left navigation pane is empty, select `File` -> `Sync Project with Gradle Files` and try to re-run the project by selecting `Run` -> `Run 'All Tests'`.
```
